pkgs/sops-install-secrets/main.go

```diff
@@ -102,6 +103,30 @@ type appContext struct {
 	checkMode   CheckMode
 }
 
+func secureSymlinkChown(symlinkToCheck, expectedTarget string, owner, group int) error {
+	fd, err := unix.Open(symlinkToCheck, unix.O_CLOEXEC|unix.O_PATH|unix.O_NOFOLLOW, 0)
+	if err != nil {
+			return fmt.Errorf("Failed to open %s: %w", symlinkToCheck, err)
+	}
+	defer unix.Close(fd)
+
+	buf := make([]byte, len(expectedTarget) + 1)
+	n, err := unix.Readlinkat(fd, "", buf)
```

Comment:
If n is larger than `symlinkToCheck` than there might be truncation happening. This was the actual reason I added + 1 in https://github.com/Mic92/sops-nix/pull/36/files#diff-60efc9b29f1818e86a75e498ef70348aR136
In https://github.com/Mic92/sops-nix/pull/36/files#diff-60efc9b29f1818e86a75e498ef70348aR138
I check if the same: https://github.com/Mic92/sops-nix/pull/36/files#diff-60efc9b29f1818e86a75e498ef70348aR138

Reply:
I think I get it now. I'll update this a smidge and fix the comment.